packages/framework-core/src/services/read-model-store.ts

```diff
@@ -94,7 +95,8 @@ export class ReadModelStore {
     projectionMetadata: ProjectionMetadata<EntityInterface>,
     readModelName: string,
     readModelID: UUID,
-    sequenceKey?: SequenceKey
+    sequenceKey?: SequenceKey,
+    entityTypeName?: string
   ): Promise<unknown> {
```

Comment:
In oder to have the information I proposed in the other comment, you would need to pass here the `entitySnapshotEnvelope`, instead of only the entityTypeName